src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1187,6 +1187,46 @@ private static DateTime ConvertToUTC(DateTime dateTime, TimeZoneInfo fromTimeZon
             return resultDateTime.UtcDateTime;
         }
 
+        public static FormulaValue Unichar(IRContext irContext, NumberValue[] args)
+        {
+            double number = args[0].Value;
+
+            // Unicode upper and lower bounds
+            const double lowerBound = 1; // Excel returns error for Unichar(0)
+            const double upperBound = 1114111; // 0x10FFFF
+
+            // https://unicode.org/glossary/#surrogate_pair
+            const double surrogateMin = 55296; // 0xD800
+            const double surrogateMax = 57343; // 0xDFFF
+
+            bool isOutOfRange = number < lowerBound || number > upperBound;
+            bool isPartialSurrogate = number >= surrogateMin && number <= surrogateMax;
+
+            if (isOutOfRange)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = $"Input value {number} falls outside the allowable range",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.InvalidArgument
+                });
+            }
+            else if (isPartialSurrogate)

```

Comment:
No need to use `else` here, since the previous block returns already

Reply:
Do we need to separate "> 0x10FFFF" and [0xD800-0xDFFF] cases?

Maybe a sinple test/error/message is enough

Reply:
@LucGenetier just being consistent with the Excel behavior , as Excel Unichar returns "#VALUE" and "#NA" errors